source/server/http/admin.cc

```diff
@@ -514,6 +514,29 @@ Http::Code AdminImpl::handlerCpuProfiler(absl::string_view url, Http::HeaderMap&
   return Http::Code::OK;
 }
 
+Http::Code AdminImpl::handlerHeapProfiler(absl::string_view url, Http::HeaderMap&,
+                                          Buffer::Instance& response, AdminStream&) {
+  Http::Utility::QueryParams query_params = Http::Utility::parseQueryString(url);
+  if (query_params.size() != 1 || query_params.begin()->first != "enable" ||
+      (query_params.begin()->second != "y" && query_params.begin()->second != "n")) {
+    response.add("?enable=<y|n>\n");
+    return Http::Code::BadRequest;
+  }
+
+  bool enable = query_params.begin()->second == "y";
+  if (enable && !Profiler::Heap::profilerEnabled()) {
+    if (!Profiler::Heap::startProfiler(profile_path_)) {
+      response.add("failure to start the heap profiler");
+      return Http::Code::InternalServerError;
+    }
+  } else if (!enable && Profiler::Heap::profilerEnabled()) {
+    Profiler::Heap::stopProfiler();
+  }
+
+  response.add("OK\n");
```

Comment:
Could we also add a brief note -- maybe just a URL pointing to doc -- indicating how to access the profile data?

Reply:
Great idea. Will do!